binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java

```diff
@@ -0,0 +1,241 @@
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+
+import org.springframework.beans.factory.annotation.AnnotatedBeanDefinition;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.ResolvableType;
+import org.springframework.kafka.support.serializer.JsonSerde;
+import org.springframework.util.ClassUtils;
+
+/**
+ * Utility class that contains various methods to help resolve {@link Serde Serdes}.
+ *
+ * @author Chris Bono
+ * @since 4.0
+ */
+abstract class SerdeResolverUtils {
+
+	private static final Log LOG = LogFactory.getLog(SerdeResolverUtils.class);
+
+	/**
+	 *  Return the closest matching configured {@code Serde<?>} bean if one exists, or the specified
+	 *  {@code fallbackSerde}, or finally a standard default serde if no fallback specified.
+	 *
+	 * @param context the application context
+	 * @param targetType the target type to find the serde for
+	 * @param fallbackSerde the fallback serde in case no matching serde bean found in the context
+	 * @return serde to use for the target type
+	 */
+	static Serde<?> resolveForType(ConfigurableApplicationContext context, ResolvableType targetType, Serde<?> fallbackSerde) {
+
+		List<Serde<?>> matchingSerdes = findMatchingSerdes(context, targetType);
+		if (!matchingSerdes.isEmpty()) {
+			return matchingSerdes.get(0);
+		}
+
+		// We don't attempt to find a matching Serde for type '?'
+		if (targetType.getRawClass() == null) {
+			return null;
+		}
+
+		Serde<?> serde = null;
+		Class<?> genericRawClazz = targetType.getRawClass();
+		if (Integer.class.isAssignableFrom(genericRawClazz)) {
+			serde = Serdes.Integer();
+		}
+		else if (Long.class.isAssignableFrom(genericRawClazz)) {
+			serde = Serdes.Long();
+		}
+		else if (Short.class.isAssignableFrom(genericRawClazz)) {
+			serde = Serdes.Short();
+		}
+		else if (Double.class.isAssignableFrom(genericRawClazz)) {
+			serde = Serdes.Double();
+		}
+		else if (Float.class.isAssignableFrom(genericRawClazz)) {
+			serde = Serdes.Float();
+		}
+		else if (byte[].class.isAssignableFrom(genericRawClazz)) {
+			serde = Serdes.ByteArray();
+		}
+		else if (String.class.isAssignableFrom(genericRawClazz)) {
+			serde = Serdes.String();
+		}
+		else if (UUID.class.isAssignableFrom(genericRawClazz)) {
+			serde = Serdes.UUID();
+		}
+		else if (!isSerdeFromStandardDefaults(fallbackSerde)) {
+			// User purposely set a default serde that is not one of the above
+			serde = fallbackSerde;
+		}
+		else {
+			// If the type is Object, then skip assigning the JsonSerde and let the fallback mechanism takes precedence.
+			if (!genericRawClazz.isAssignableFrom((Object.class))) {
+				serde = new JsonSerde(genericRawClazz);
+			}
+		}
+		return serde;
+	}
+
+	private static boolean isSerdeFromStandardDefaults(Serde<?> serde) {
+		if (serde != null) {
```

Comment:
**[UNRELATED]** I am going to do a follow-on PR to rework this default fallback portion of the code
- invert the if (short circuit) 
- slightly change how the "is default serdes" is implemented

Reply:
To be handled in #2370